src/switch_core_io.c

```diff
@@ -194,10 +194,33 @@ SWITCH_DECLARE(switch_status_t) switch_core_session_read_frame(switch_core_sessi
 		}
 
 		if (status == SWITCH_STATUS_INUSE) {
+			switch_bool_t has_bug_with_continue_on_hold = SWITCH_FALSE;
 			*frame = &runtime.dummy_cng_frame;
 			if (switch_test_flag(*frame, SFF_FORK_RTP)) {
 				switch_core_session_set_fork_read_frame(session, *frame);
 			}
+
+			switch_thread_rwlock_rdlock(session->bug_rwlock);
+			for (switch_media_bug_t *bp = session->bugs; bp; bp = bp->next) {
+				if (switch_core_media_bug_test_flag(bp, SMBF_CONTINUE_ON_HOLD)) {
+					has_bug_with_continue_on_hold = SWITCH_TRUE;
+					break;
+				}
+			}
+			switch_thread_rwlock_unlock(session->bug_rwlock);
+			if (has_bug_with_continue_on_hold) {
+				switch_mutex_lock(session->codec_read_mutex);
+				if (!switch_core_codec_ready(session->read_codec)) {
+					switch_mutex_unlock(session->codec_read_mutex);
+					switch_yield(20000);
+					return SWITCH_STATUS_SUCCESS;
+				}
+				(*frame)->codec = session->read_codec;
+				switch_mutex_unlock(session->codec_read_mutex);
+				status = SWITCH_STATUS_SUCCESS;
+				cng_frame = 1;
```

Comment:
probably a better name for this flag, should this be something like is_hold or is_inuse?

Reply:
got it